### PR TITLE
Remove legacy stacksampler logic

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
/cc @jkotas

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

File: src/coreclr/inc/clrconfigvalues.h [hunk 1]
```diff
@@ -320,13 +320,6 @@ RETAIL_CONFIG_STRING_INFO(EXTERNAL_AltJitOs, W("AltJitOS"), "Sets target OS for
...
 #endif // defined(ALLOW_SXS_JIT)
 
-#if defined(FEATURE_STACK_SAMPLING)
-RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_StackSamplingEnabled, W("StackSamplingEnabled"), 0, "Is stack sampling based tracking of evolving hot methods enabled.")
-RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_StackSamplingAfter, W("StackSamplingAfter"), 0, "When to start sampling (for some sort of app steady state), i.e., initial delay for sampling start in milliseconds.")
-RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_StackSamplingEvery, W("StackSamplingEvery"), 100, "How frequent should thread stacks be sampled in milliseconds.")
-RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_StackSamplingNumMethods, W("StackSamplingNumMethods"), 32, "Number of evolving methods to track as hot and JIT them in the background at a given point of execution.")
-#endif // defined(FEATURE_JIT_SAMPLING)
-
 #if defined(ALLOW_SXS_JIT_NGEN)
```
Comment (jkotas) [inline src/coreclr/inc/clrconfigvalues.h:323]: `ALLOW_SXS_JIT_NGEN` is dead code as well.

File: src/coreclr/vm/jitinterface.cpp [hunk 2]
```diff
@@ -13102,11 +13085,7 @@ PCODE UnsafeJitFunction(PrepareCodeConfig* config,
             // Note: if we're only importing (ie, verifying/
```
Comment (jkotas) [inline src/coreclr/vm/jitinterface.cpp:13085]: This comment does not make sense. (It did not make sense even before this change.)

File: src/coreclr/vm/jitinterface.cpp [hunk 3]
```diff
@@ -13102,11 +13085,7 @@ PCODE UnsafeJitFunction(PrepareCodeConfig* config,
...
             // checking to make sure we could JIT, but not actually generating code (
             // eg, for inlining), then DON'T TELL THE DEBUGGER about this.
```
Comment (jkotas) [inline src/coreclr/vm/jitinterface.cpp:13087]:
```suggestion
```

**Review by davidwrighton — APPROVED** [review]

**Pull request merged**